code/game/objects/effects/spawners/random/trash.dm

```diff
@@ -266,3 +266,22 @@
 		/obj/item/food/breadslice/moldy/bacteria,
 		/obj/item/food/pizzaslice/moldy/bacteria,
 	)
+
+/obj/effect/spawner/random/trash/crushed_can
+	name = "crushed can spawner"
+	icon_state = "crushed_can"
+	loot = list(/obj/item/trash/can)
+	/// Whether the can will spawn with this spawner's icon_state instead of a random one (used for mapedits)
+	var/random_icon = TRUE
```

Comment:
Just remove this variable. If someone wants a crushed can with a specific icon state, they could simply make a subtype with that state or at worst mapedit the object itself at worst.